addons/website_sale/models/website.py

```diff
@@ -59,7 +59,7 @@ def _get_pl_partner_order(self, country_code, show_visible, website_pl, current_
         pricelists = self.env['product.pricelist']
         if country_code:
             for cgroup in self.env['res.country.group'].search([('country_ids.code', '=', country_code)]):
-                for group_pricelists in cgroup.pricelist_ids:
+                for group_pricelists in cgroup.pricelist_ids.filtered(lambda pl: pl.website_id.id in (False, self.id)):
```

Comment:
Can't you add this in the `search` instead of filtering afterwards?

Reply:
I don't think so. By searching `country.group` for the given `country_code` and having a pricelist 'website available' , I would still have to refilter the pricelist_ids afterwards as a `res.country.group` might have at least one pricelist 'website available' without especially having all its pricelists 'website available'.

Plus, I would have to mimick the `_get_website_pricelists_domain()` method (that is a domain supposed to be used on `product.pricelist` directly.

Here, to add the domain in the search, I would have to copy the 'accessible on website' condition that I would like to keep encapsulated only in `_get_website_pricelists_domain()` (and its filter equivalent `_is_available_on_website()`).

Currently, it is:
```python
if country_code:
    for cgroup in self.env['res.country.group'].search([('country_ids.code', '=', country_code)]):
        pricelists |= cgroup.pricelist_ids.filtered(lambda pl: pl._is_available_on_website(self.id) and _check_show_visible(pl))
```
If I add the domain in search, it would become:
```python
if country_code:
    domain = [
        ('country_ids.code', '=', country_code),
        '|', ('pricelist_ids.website_id', '=', website_id),
        '&', ('website_id', '=', False),
        '|', ('selectable', '=', True), ('code', '!=', False),
    ]
    for cgroup in self.env['res.country.group'].search([('country_ids.code', '=', country_code)]):
        pricelists |= cgroup.pricelist_ids.filtered(lambda pl: pl._is_available_on_website(self.id) and _check_show_visible(pl))
```

But maybe it is worth speeding a little bit the search with the domain even if we need to refilter in python afterward?